firewood/src/merkle/parallel.rs

```diff
@@ -0,0 +1,500 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+use crate::db::BatchOp;
+use crate::merkle::{Merkle, Value};
+use crate::v2::api::KeyValuePairIter;
+use firewood_storage::{
+    BranchNode, Child, FileBacked, FileIoError, ImmutableProposal, LeafNode, MutableProposal,
+    NibblesIterator, Node, NodeStore, Parentable, Path,
+};
+use rayon::{ThreadPool, ThreadPoolBuilder};
+use std::iter::once;
+use std::sync::mpsc::{Receiver, Sender};
+use std::sync::{Arc, OnceLock, mpsc};
+
+#[derive(Debug)]
+struct WorkerState {
+    sender: mpsc::Sender<Request>,
+}
```

Comment:
This is just a newtype pattern I think:
```suggestion
struct WorkerSender(mpsc::Sender<Request>);

impl std::ops::Deref for WorkerSender {
    type Target = mpsc::Sender<Request>;
    
    fn deref(&self) -> &Self::Target {
        &self.0
    }
}
```
Then, if you have a WorkerSender you can just call send on it (no reason to name sender).
    

```